### PR TITLE
Named export for macros plugin

### PR DESCRIPTION
**Pull request by pcattori**

### CODE REVIEW

**Review by markdalgleish — COMMENTED** [review]
We should update the package description from `Minimal Vite plugin for for isolating server-only and client-only code` to something like `A collection of Vite plugins for isolating server-only and client-only code`.

Also, the readme currently says `for for isolating` 🙈

File: src/index.ts [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
-
+export function envOnlyMacros(): PluginOption[] {
   return [
     {
-      name: "vite-plugin-env-only",
-      configResolved(config) {
-        root = config.root
-        command = config.command
-      },
+      name: "vite-plugin-env-only-macros",
```
Comment (markdalgleish) [inline src/index.ts:11]: Nit: Can we shorten this to `vite-env-only-macros` (and the other plugin names too)?

**Review by markdalgleish — APPROVED** [review]
A couple of minor things, but this looks great.

File: src/index.ts (hunk 1 above)
Comment (pcattori) [reply src/index.ts:11]: From the [Vite Plugin API docs](https://vitejs.dev/guide/api-plugin#conventions)

> Vite Plugins should have a clear name with `vite-plugin-` prefix.

Though its unclear if they mean the package name or the plugin name (I think the latter)

Comment (pcattori) [reply src/index.ts:11]: Also, `@vitejs/plugin-react` doesn't follow this convention, (neither in package name nor plugin names). Though maybe this convention is for 3rd party plugins where the `@vitejs` namespace wouldn't be present. So that makes me think its a package name convention?

Comment (pcattori) [reply src/index.ts:11]: `vite-tsconfig-paths` doesn't follow this convention for the package name nor the plugin name either 🤷
